plan/property_cols_prune.go

```diff
@@ -58,9 +116,16 @@ func (p *LogicalJoin) preparePossibleProperties() [][]*expression.Column {
 	} else if p.JoinType == RightOuterJoin {
 		leftProperties = nil
 	}
-	resultProperties := make([][]*expression.Column, len(leftProperties), len(leftProperties)+len(rightProperties))
-	copy(resultProperties, leftProperties)
-	resultProperties = append(resultProperties, rightProperties...)
+	resultProperties := make([][]*expression.Column, len(leftProperties)+len(rightProperties))
```

Comment:
Why make this change?

Reply:
It's a slice of slice. So when do the copy on `resultProperties`, the inner element still shares the same slice. If there's `Projection->Join`, The change of `Projection`'s prop will have influence on `Join`'s prop.

Reply:
Does there any test case that covered this?

Reply:
Without this, two of the existing unit-test will fail.